src/test/java/tr/com/srdc/cda2fhir/DeduplicationTest.java

```diff
@@ -0,0 +1,131 @@
+package tr.com.srdc.cda2fhir;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Medication;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openhealthtools.mdht.uml.cda.Component3;
+import org.openhealthtools.mdht.uml.cda.SubstanceAdministration;
+import org.openhealthtools.mdht.uml.cda.consol.ContinuityOfCareDocument;
+import org.openhealthtools.mdht.uml.cda.consol.MedicationActivity;
+import org.openhealthtools.mdht.uml.cda.util.CDAUtil;
+
+import tr.com.srdc.cda2fhir.testutil.BundleUtil;
+import tr.com.srdc.cda2fhir.testutil.CDAFactories;
+import tr.com.srdc.cda2fhir.testutil.generator.CDAMedicationSectionComponentGenerator;
+import tr.com.srdc.cda2fhir.testutil.generator.ClinicalDocumentMetadataGenerator;
+import tr.com.srdc.cda2fhir.testutil.generator.MedicationActivityGenerator;
+import tr.com.srdc.cda2fhir.testutil.generator.MedicationInformationGenerator;
+import tr.com.srdc.cda2fhir.testutil.generator.MedicationSupplyOrderGenerator;
+import tr.com.srdc.cda2fhir.transform.CCDTransformerImpl;
+
+public class DeduplicationTest {
+	static CCDTransformerImpl ccdTransformer;
+	static CDAFactories factories;
+	static ClinicalDocumentMetadataGenerator metadataGenerator;
+	static CDAMedicationSectionComponentGenerator medSectionComponentGenerator;
+
+	@BeforeClass
+	public static void init() {
+		CDAUtil.loadPackages();
+		ccdTransformer = new CCDTransformerImpl();
+		factories = CDAFactories.init();
+		metadataGenerator = new ClinicalDocumentMetadataGenerator();
+		medSectionComponentGenerator = new CDAMedicationSectionComponentGenerator();
+	}
+
+	private MedicationActivityGenerator getMedicationActivityGeneratorOneCode(
+			MedicationInformationGenerator medInfoGenerator) {
+
```

Comment:
I am not sure why we are passing medInfoGenerator. Every single call to this gets it from defaultInstance();

Reply:
Each time MedicationInformation.getDefaultInstance is called, it calls CDGenerator.getNextInstance. Since MedicationActivity calls MedicationInformation.getDefaultInstance directly, and it's also called inside of MedicationSupplyOrder.getDefaultInstance(), CDGenerator.getNextInstance is called twice, yeilding two different medication codes: one for the consumable, and one for the medicationSupplyOrder. In many cases a medicationActivity's consumable.manufacturedProduct entry and supply.product.manufacturedProduct entries are the same drug. I thought it would be good to emulate that here, so I added the option for adding the same medicationInformation instance.

Maybe we should make a MedicationSupplyOrder constructor that takes in a MedicationInformationGenerator for this purpose? I like the option to make them different as well. I have a test for both just in case.

Reply:
right just have the medicatiionInformationeGenerator as a member and call defaultInstance inside.